deptofdefense/move.mil
# Print-Friendly PPM Estimator results

Implements #356.

## Checklist

I have…

- [x] attached screenshots illustrating relevant behavior before and after my changes in a modern desktop browser, in MSIE 11, and in a mobile browser.

## Summary of Changes

This pull request adds a print-specific stylesheet for the PPM Estimator page. This stylesheet restricts the output during printing to only the results of the PPM calculation. The results have also been expanded to show all of the inputs the user fed to the calculation, including the origin and destination (with map), the branch of service, the rank, dependents, the date, and the weights.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. git checkout `ppm-estimator-print-friendly`,
1. set up development dependencies according to [CONTRIBUTING.md](https://github.com/deptofdefense/move.mil/blob/master/CONTRIBUTING.md),
1. run `bin/rails server`, and
1. load up [http://localhost:3000](http://localhost:3000) in the Web browser of your choice.

## Screenshots

<details><summary>Chrome</summary>

![fireshot capture 72 - personally procured move ppm_ - http___localhost_3000_resources_ppm-estimator](https://user-images.githubusercontent.com/26554826/37792679-cc056d08-2de2-11e8-9d65-2ce0501f8902.png)

</details>

<details><summary>Internet Explorer 11</summary>

![virtualbox_msedge - win10_22_03_2018_15_42_02](https://user-images.githubusercontent.com/26554826/37794374-af4d2728-2de7-11e8-8cd7-067fa7592105.png)

</details>

<details><summary>Mobile Safari</summary>

![safari - mar 22 2018 at 4 24 pm](https://user-images.githubusercontent.com/26554826/37796486-d0195e8a-2ded-11e8-928f-e3bb37b74f8a.png)

</details>

<details><summary>Printed output - Chrome</summary>

[Personally Procured Move (PPM) Estimator — Move.mil (Chrome).pdf](https://github.com/deptofdefense/move.mil/files/1838901/Personally.Procured.Move.PPM.Estimator.Move.mil.Chrome.pdf)

</details>

<details><summary>Printed output - Firefox (Background colors / Background images NOT enabled)</summary>

[Personally Procured Move (PPM) Estimator — Move.mil (Firefox).pdf](https://github.com/deptofdefense/move.mil/files/1838921/Personally.Procured.Move.PPM.Estimator.Move.mil.Firefox.pdf)

</details>

<details><summary>Printed output - Safari</summary>

[Personally Procured Move (PPM) Estimator — Move.mil (Safari).pdf](https://github.com/deptofdefense/move.mil/files/1838925/Personally.Procured.Move.PPM.Estimator.Move.mil.Safari.pdf)

</details>

<details><summary>Printed output - Internet Explorer 11 (Background colors / Background Images setting in Page Settings NOT enabled)</summary>

[Personally Procured Move (PPM) Estimator - Move.mil (IE11).pdf](https://github.com/deptofdefense/move.mil/files/1839009/Personally.Procured.Move.PPM.Estimator.-.Move.mil.IE11.pdf)

</details>

<details><summary>Printed output - Edge (no way to control background images / colors)</summary>

[Personally Procured Move (PPM) Estimator - Move.mil (Edge).pdf](https://github.com/deptofdefense/move.mil/files/1839008/Personally.Procured.Move.PPM.Estimator.-.Move.mil.Edge.pdf)

</details>
